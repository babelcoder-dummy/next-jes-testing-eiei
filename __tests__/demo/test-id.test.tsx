import TestID from '@/demo/components/TestId';
import { render, screen, within } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { range } from 'lodash';

describe('Test ID', () => {
  it('renders TestId unchanged', () => {
    const { container } = render(<TestID></TestID>);

    expect(container).toMatchSnapshot();
  });

  it('renders TestId correctly', async () => {
    const user = userEvent.setup();
    render(<TestID></TestID>);

    expect(screen.getByTestId('title')).toHaveTextContent(
      'My Header with Test ID',
    );

    for (const n of range(6)) {
      const card = screen.getByTestId(`card-${n}`);

      expect(within(card).getByTestId(`product-image-${n}`)).toHaveProperty(
        'alt',
        'No Image',
      );
      expect(within(card).getByTestId(`product-title-${n}`)).toHaveTextContent(
        `Title ${n}`,
      );
      expect(within(card).getByTestId(`product-desc-${n}`)).toHaveTextContent(
        `Description ${n}`,
      );
      expect(
        within(card).getByTestId(`product-content-${n}`),
      ).toHaveTextContent(`Content ${n}`);

      await user.click(within(card).getByTestId(`product-details-${n}`));
      expect(screen.getByTestId(`dialog-title-${n}`)).toHaveTextContent(
        'Hey guy!',
      );
      expect(screen.getByTestId(`dialog-desc-${n}`)).toHaveTextContent(
        `The product ID is ${n}`,
      );
      await user.click(screen.getByTestId(`dialog-close-${n}`));
    }
  });
});
