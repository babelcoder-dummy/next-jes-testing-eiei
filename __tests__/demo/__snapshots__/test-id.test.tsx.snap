// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ID renders TestId unchanged 1`] = `
<div>
  <section>
    <header>
      <h2
        class="text-center text-4xl font-bold"
        data-testid="title"
      >
        My Header with Test ID
      </h2>
    </header>
    <div
      class="shrink-0 bg-border h-[1px] w-full my-4"
      data-orientation="horizontal"
      role="none"
    />
    <div
      class="grid grid-cols-3 gap-4"
    >
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm"
        data-testid="card-0"
      >
        <div
          class="relative h-40"
        >
          <img
            alt="No Image"
            class="object-cover"
            data-nimg="fill"
            data-testid="product-image-0"
            decoding="async"
            fetchpriority="high"
            sizes="(min-width: 808px) 50vw, 100vw"
            src="/_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=3840&q=75"
            srcset="/_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=384&q=75 384w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=640&q=75 640w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=750&q=75 750w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=828&q=75 828w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        <div
          class="flex flex-col space-y-1.5 p-6"
        >
          <h3
            class="text-2xl font-semibold leading-none tracking-tight"
            data-testid="product-title-0"
          >
            Title 
            0
          </h3>
          <p
            class="text-sm text-muted-foreground"
            data-testid="product-desc-0"
          >
            Description 
            0
          </p>
        </div>
        <div
          class="p-6 pt-0"
        >
          <p
            data-testid="product-content-0"
          >
            Content 
            0
          </p>
        </div>
        <div
          class="flex items-center p-6 pt-0"
        >
          <button
            aria-controls="radix-:r0:"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 px-4 py-2"
            data-state="closed"
            data-testid="product-details-0"
            type="button"
          >
            Details
          </button>
        </div>
      </div>
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm"
        data-testid="card-1"
      >
        <div
          class="relative h-40"
        >
          <img
            alt="No Image"
            class="object-cover"
            data-nimg="fill"
            data-testid="product-image-1"
            decoding="async"
            fetchpriority="high"
            sizes="(min-width: 808px) 50vw, 100vw"
            src="/_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=3840&q=75"
            srcset="/_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=384&q=75 384w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=640&q=75 640w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=750&q=75 750w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=828&q=75 828w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        <div
          class="flex flex-col space-y-1.5 p-6"
        >
          <h3
            class="text-2xl font-semibold leading-none tracking-tight"
            data-testid="product-title-1"
          >
            Title 
            1
          </h3>
          <p
            class="text-sm text-muted-foreground"
            data-testid="product-desc-1"
          >
            Description 
            1
          </p>
        </div>
        <div
          class="p-6 pt-0"
        >
          <p
            data-testid="product-content-1"
          >
            Content 
            1
          </p>
        </div>
        <div
          class="flex items-center p-6 pt-0"
        >
          <button
            aria-controls="radix-:r3:"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 px-4 py-2"
            data-state="closed"
            data-testid="product-details-1"
            type="button"
          >
            Details
          </button>
        </div>
      </div>
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm"
        data-testid="card-2"
      >
        <div
          class="relative h-40"
        >
          <img
            alt="No Image"
            class="object-cover"
            data-nimg="fill"
            data-testid="product-image-2"
            decoding="async"
            fetchpriority="high"
            sizes="(min-width: 808px) 50vw, 100vw"
            src="/_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=3840&q=75"
            srcset="/_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=384&q=75 384w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=640&q=75 640w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=750&q=75 750w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=828&q=75 828w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        <div
          class="flex flex-col space-y-1.5 p-6"
        >
          <h3
            class="text-2xl font-semibold leading-none tracking-tight"
            data-testid="product-title-2"
          >
            Title 
            2
          </h3>
          <p
            class="text-sm text-muted-foreground"
            data-testid="product-desc-2"
          >
            Description 
            2
          </p>
        </div>
        <div
          class="p-6 pt-0"
        >
          <p
            data-testid="product-content-2"
          >
            Content 
            2
          </p>
        </div>
        <div
          class="flex items-center p-6 pt-0"
        >
          <button
            aria-controls="radix-:r6:"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 px-4 py-2"
            data-state="closed"
            data-testid="product-details-2"
            type="button"
          >
            Details
          </button>
        </div>
      </div>
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm"
        data-testid="card-3"
      >
        <div
          class="relative h-40"
        >
          <img
            alt="No Image"
            class="object-cover"
            data-nimg="fill"
            data-testid="product-image-3"
            decoding="async"
            fetchpriority="high"
            sizes="(min-width: 808px) 50vw, 100vw"
            src="/_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=3840&q=75"
            srcset="/_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=384&q=75 384w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=640&q=75 640w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=750&q=75 750w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=828&q=75 828w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        <div
          class="flex flex-col space-y-1.5 p-6"
        >
          <h3
            class="text-2xl font-semibold leading-none tracking-tight"
            data-testid="product-title-3"
          >
            Title 
            3
          </h3>
          <p
            class="text-sm text-muted-foreground"
            data-testid="product-desc-3"
          >
            Description 
            3
          </p>
        </div>
        <div
          class="p-6 pt-0"
        >
          <p
            data-testid="product-content-3"
          >
            Content 
            3
          </p>
        </div>
        <div
          class="flex items-center p-6 pt-0"
        >
          <button
            aria-controls="radix-:r9:"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 px-4 py-2"
            data-state="closed"
            data-testid="product-details-3"
            type="button"
          >
            Details
          </button>
        </div>
      </div>
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm"
        data-testid="card-4"
      >
        <div
          class="relative h-40"
        >
          <img
            alt="No Image"
            class="object-cover"
            data-nimg="fill"
            data-testid="product-image-4"
            decoding="async"
            fetchpriority="high"
            sizes="(min-width: 808px) 50vw, 100vw"
            src="/_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=3840&q=75"
            srcset="/_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=384&q=75 384w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=640&q=75 640w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=750&q=75 750w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=828&q=75 828w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        <div
          class="flex flex-col space-y-1.5 p-6"
        >
          <h3
            class="text-2xl font-semibold leading-none tracking-tight"
            data-testid="product-title-4"
          >
            Title 
            4
          </h3>
          <p
            class="text-sm text-muted-foreground"
            data-testid="product-desc-4"
          >
            Description 
            4
          </p>
        </div>
        <div
          class="p-6 pt-0"
        >
          <p
            data-testid="product-content-4"
          >
            Content 
            4
          </p>
        </div>
        <div
          class="flex items-center p-6 pt-0"
        >
          <button
            aria-controls="radix-:rc:"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 px-4 py-2"
            data-state="closed"
            data-testid="product-details-4"
            type="button"
          >
            Details
          </button>
        </div>
      </div>
      <div
        class="rounded-lg border bg-card text-card-foreground shadow-sm"
        data-testid="card-5"
      >
        <div
          class="relative h-40"
        >
          <img
            alt="No Image"
            class="object-cover"
            data-nimg="fill"
            data-testid="product-image-5"
            decoding="async"
            fetchpriority="high"
            sizes="(min-width: 808px) 50vw, 100vw"
            src="/_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=3840&q=75"
            srcset="/_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=384&q=75 384w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=640&q=75 640w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=750&q=75 750w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=828&q=75 828w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fassets%2Fimages%2Fno-image.jpg&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
        </div>
        <div
          class="flex flex-col space-y-1.5 p-6"
        >
          <h3
            class="text-2xl font-semibold leading-none tracking-tight"
            data-testid="product-title-5"
          >
            Title 
            5
          </h3>
          <p
            class="text-sm text-muted-foreground"
            data-testid="product-desc-5"
          >
            Description 
            5
          </p>
        </div>
        <div
          class="p-6 pt-0"
        >
          <p
            data-testid="product-content-5"
          >
            Content 
            5
          </p>
        </div>
        <div
          class="flex items-center p-6 pt-0"
        >
          <button
            aria-controls="radix-:rf:"
            aria-expanded="false"
            aria-haspopup="dialog"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 px-4 py-2"
            data-state="closed"
            data-testid="product-details-5"
            type="button"
          >
            Details
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;
